implementation.d.ts

```diff
@@ -0,0 +1,54 @@
+interface PromiseResolution<T> {
+	status: 'fulfilled';
+	value: T;
+}
+
+interface PromiseRejection<E extends Error = Error> {
+	status: 'rejected';
+	reason: E;
+}
+
+type PromiseResult<T, E extends Error = Error> = PromiseResolution<T> | PromiseRejection<E>;
+
+type PromiseTuple1<T1> = [Promise<T1>];
```

Comment:
is writing out all the tuples really the best way typescript has to handle this?

Reply:
That's a matter of opinion, I would say. It's one possible way, I do not insist on writing the code this way. The tuple types could be inlined in the function declaration and it would work just fine.

Reply:
i mean, is there no way to avoid any of them, and just say “all the types passed in, map to these passed out”?

Reply:
After more research, I was able to find this: https://github.com/Microsoft/TypeScript/pull/26063

To be honest, I was not able to find a solution to this previously, so I went with an approach I saw in the reselect library.

I will fix this, thanks for bringing this up.